src/sentry/workflow_engine/models/workflow.py

```diff
@@ -61,6 +61,7 @@ def evaluate_trigger_conditions(self, job: WorkflowJob) -> bool:
         if self.when_condition_group is None:
             return True
 
+        job["workflow"] = self
```

Comment:
nice

Reply:
🤔 should we do this in the process_workflows method so it's available on the job even if the workflow hasn't been evaluated?

Reply:
@saponifi3d it's not used until we evaluate conditions and we need to pass each workflow individually into the same job 🤔

Reply:
i'm also thinking of consistency with the job data class. this method is setup to only evaluate data, not mutate the job, so it's a side effect from this methods perspective. Some things I think about are; What if we want to evaluate this model multiple times? Should it always set the workflow on the job or does that only need to happen once?

Reply:
@saponifi3d are you saying we should do this outside like here? https://github.com/getsentry/sentry/blob/c563acb7c659f6814ea48f5d0bc0e44eaed8ba44/src/sentry/workflow_engine/processors/workflow.py#L18-L19